internal/providers/azure/azure.go

```diff
@@ -66,29 +64,52 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 // FetchFromOvfDevice has the return signature of platform.NewFetcher. It is
 // wrapped by this and AzureStack packages.
 func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config, report.Report, error) {
-	devicePath := filepath.Join(distro.DiskByIDDir(), configDeviceID)
-
 	logger := f.Logger
-	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
 
-	fsType, err := checkOvfFsType(logger, devicePath, ovfFsTypes)
-	if err != nil {
-		return types.Config{}, report.Report{}, err
+	for _, ovfFsType := range ovfFsTypes {
+		devices, err := execUtil.GetBlockDevices(ovfFsType)
+		if err != nil {
+			return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve block devices with FSTYPE=%q: %v", ovfFsType, err)
+		}
+		for len(devices) == 0 {
```

Comment:
This still assumes that if any devices of type `ovfFsType` show up, no additional such devices will show up later.  We can't assume that.  We should go through the `ovfFsTypes` loop without any retries, and if we don't find any candidate devices, sleep and retry the entire loop.

For extra cleanliness, we can keep a map of device paths we've seen before, and not try mounting them again on subsequent iterations if we've already tried once.